storage/remote/queue_manager.go

```diff
@@ -1021,19 +1021,15 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan interface
 		// TODO(cstyan): Casting this many times smells, also we could get index out of bounds issues here.
 		maxExemplars                                 = int(math.Max(1, float64(max/10)))
 		nPending, nPendingSamples, nPendingExemplars = 0, 0, 0
-		sampleBuffer                                 = allocateSampleBuffer(max)
 
-		buf            []byte
-		pendingData    []prompb.TimeSeries
-		exemplarBuffer [][]prompb.Exemplar
+		buf         []byte
+		pendingData []prompb.TimeSeries
 	)
-	totalPending := max
 	if s.qm.sendExemplars {
-		exemplarBuffer = allocateExemplarBuffer(maxExemplars)
-		totalPending += maxExemplars
+		max += maxExemplars
```

Comment:
I'm wondering if we should make this just `max += max*0.1` or have the increased amount be configurable. The other option is to just document a suggestion of: 
> increase your `max_samples_per_send by X% if you have send_exemplars enabled`

I'm leaning towards the last option, as we have a lot of configurable knobs already.

@csmarchbanks thoughts?

Reply:
Hmm, I do agree that I do not want additional configuration options. I would prefer not to document that suggestion yet as I wonder if we could do something better than counting samples and exemplars separately. I would think that much of the time when sending exemplars they will be added close to the sample for that same series and we could combine exemplars and samples into the same series.

TLDR I would say just do `max += max*0.1` if exemplars are enabled for now.

Reply:
In other words, will we even need max += after https://github.com/prometheus/prometheus/pull/9075 is merged?

Reply:
> In other words, will we even need max += after #9075 is merged?

Exactly. No, I don't think so.

Reply:
I'll make it `max += max*0.1`, although I wonder if we should document that just so users know why extra samples are being sent. Then again it seems that it probably won't make it past #9075 so maybe it doesn't matter.